envoy/ssl/certificate_validation_context_config.h

```diff
@@ -89,6 +89,11 @@ class CertificateValidationContextConfig {
    * @return whether to validate certificate chain with all CRL or not.
    */
   virtual bool onlyVerifyLeafCertificateCrl() const PURE;
+
+  /**
+   * @return the max depth used when verifying the certificate-chain
+   */
+  virtual const absl::optional<uint32_t> maxVerifyDepth() const PURE;
```

Comment:
nit: returning a const value type isn't typically done.

```suggestion
  virtual absl::optional<uint32_t> maxVerifyDepth() const PURE;
```